### PR TITLE
tidb-in-kubernetes: add note for concurrent scale-out and scale-in

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

### What is changed, added or deleted? 
Add note for concurrent scale-out and scale-in in case of miss-operation.

### Which version does your change affect? 
`dev` `3.0`

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

@shuijing198799 @weekface @lilin90 @dcalvin PTAL

### CODE REVIEW

File: dev/tidb-in-kubernetes/scale-in-kubernetes.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ TiDB 水平扩缩容操作指的是通过增加或减少节点的数量，来达
 > - PD、TiKV 组件在滚动升级的过程中不会触发扩缩容操作。
 > - PD、TiKV 组件在缩容过程中都会调用接口下线正在删除的 PD、TiKV 组件在下线时会涉及到数据迁移的操作，所以会消耗比较长的时间。
 > - PD、TiKV 组件在缩容过程中被删除的节点的 PVC 会保留，并且由于 PV 的 `Reclaim Policy` 设置为 `Retain`，即使 PVC 被删除，数据依然可以找回。
+> - TiKV 组件不支持在缩容未完成时进行扩容操作，强制执行此操作可能导致集群状态异常。假如异常已经发生，可以参考 [TiKV Store 异常进入 Tombstone 状态](tidb-in-kubernetes/troubleshoot.md#tikv-store-异常进入-tombstone-状态) 进行解决。
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/scale-in-kubernetes.md]:
```suggestion
> - TiKV 组件不支持在缩容过程中进行扩容操作，强制执行此操作可能导致集群状态异常。假如异常已经发生，可以参考 [TiKV Store 异常进入 Tombstone 状态](tidb-in-kubernetes/troubleshoot.md#tikv-store-异常进入-tombstone-状态) 进行解决。
```

File: dev/tidb-in-kubernetes/troubleshoot.md [hunk 2]
```diff
@@ -259,3 +259,41 @@ kubectl logs -f <tidb-pod-name> -n <namespace>
     * 检查 node 上的相关 route 规则是否正确
     * 检查网络插件服务是否正常
     * 参考上面的 [Pod 之间网络不通](#pod-之间网络不通)章节
+
+## TiKV Store 异常进入 Tombstone 状态
+
+正常情况下，当 TiKV Pod 处于健康状态时（Pod 状态为 `Running`），对应的 TiKV Store 状态也是健康的（Store 状态为 `UP`）。但并发操作 TiKV 组件的扩容和缩容可能会导致部分 TiKV Store 异常进入 Tombstone 状态。此时，可以按照以下步骤进行修复：
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/troubleshoot.md]:
```suggestion
正常情况下，当 TiKV Pod 处于健康状态时（Pod 状态为 `Running`），对应的 TiKV Store 状态也是健康的（Store 状态为 `UP`）。但并发进行 TiKV 组件的扩容和缩容可能会导致部分 TiKV Store 异常并进入 Tombstone 状态。此时，可以按照以下步骤进行修复：
```

File: dev/tidb-in-kubernetes/troubleshoot.md [hunk 3]
```diff
@@ -259,3 +259,41 @@ kubectl logs -f <tidb-pod-name> -n <namespace>
...
+
+2. 查看 TiKV Pod 运行状态：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl get -n <namespace> po -l app.kubernetes.io/component=tikv
+    ```
+
+3. 对比 Store 状态与 Pod 运行状态。正常情况下每一个正常运行的 Pod 都有一个处于 `UP` 状态的 Store，假如某个 TiKV 的 Pod 的所有 Store 都处于 `Tombstone` 或 `Down` 状态，则其中 `lastHeartbeatTime` 最大的 Store 即为异常下线的 Store，我们需要手工进行恢复：
```
Comment (lilin90) [inline dev/tidb-in-kubernetes/troubleshoot.md]:
```suggestion
3. 对比 Store 状态与 Pod 运行状态。正常情况下每一个正常运行的 Pod 都有一个处于 `UP` 状态的 Store，假如某个 TiKV 的 Pod 的所有 Store 都处于 `Tombstone` 或 `Down` 状态，则其中 `lastHeartbeatTime` 最大的 Store 即为异常下线的 Store，需要手工进行恢复：
```

File: v3.0/tidb-in-kubernetes/scale-in-kubernetes.md [hunk 4]
```diff
@@ -1,7 +1,6 @@
 ---
 title: Kubernetes 上的 TiDB 集群扩缩容
 category: how-to
-aliases: ['/docs-cn/v3.0/how-to/scale/tidb-in-kubernetes/']
```
Comment (lilin90) [inline v3.0/tidb-in-kubernetes/scale-in-kubernetes.md]: This is for users who save the original link. We can keep it so those users can visit the document successfully.

File: dev/tidb-in-kubernetes/troubleshoot.md [hunk 5]
```diff
@@ -259,3 +259,41 @@ kubectl logs -f <tidb-pod-name> -n <namespace>
...
+## TiKV Store 异常进入 Tombstone 状态
+
+正常情况下，当 TiKV Pod 处于健康状态时（Pod 状态为 `Running`），对应的 TiKV Store 状态也是健康的（Store 状态为 `UP`）。但并发操作 TiKV 组件的扩容和缩容可能会导致部分 TiKV Store 异常进入 Tombstone 状态。此时，可以按照以下步骤进行修复：
+
+1. 查看 TiKV Store 状态：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl get -n <namespace> tidbcluster <cluster-name> -ojson | jq '.status.tikv.stores'
```
Comment (tennix) [inline dev/tidb-in-kubernetes/troubleshoot.md]:
```suggestion
    kubectl get -n <namespace> tidbcluster <release-name> -ojson | jq '.status.tikv.stores'
```

File: dev/tidb-in-kubernetes/troubleshoot.md [hunk 6]
```diff
@@ -259,3 +259,41 @@ kubectl logs -f <tidb-pod-name> -n <namespace>
...
+    ```
+
+3. 对比 Store 状态与 Pod 运行状态。正常情况下每一个正常运行的 Pod 都有一个处于 `UP` 状态的 Store，假如某个 TiKV 的 Pod 的所有 Store 都处于 `Tombstone` 或 `Down` 状态，则其中 `lastHeartbeatTime` 最大的 Store 即为异常下线的 Store，我们需要手工进行恢复：
+
+    * 打开到 PD 服务的连接：
+
+        {{< copyable "shell-regular" >}}
+
+        ```shell
+        kubectl port-forward -n <namespace> svc/<cluster-name>-pd <local-port>:2379 &>/tmp/portforward-pd.log
```
Comment (tennix) [inline dev/tidb-in-kubernetes/troubleshoot.md]:
```suggestion
        kubectl port-forward -n <namespace> svc/<release-name>-pd <local-port>:2379 &>/tmp/portforward-pd.log
```

**Comment by aylei** [comment]
@dcalvin @lilin90 @tennix All comments addressed, PTAL

File: dev/tidb-in-kubernetes/troubleshoot.md [hunk 7]
```diff
@@ -259,3 +259,69 @@ kubectl logs -f <tidb-pod-name> -n <namespace> -c tidb
...
+
+    * 上线对应 Store：
+
+        {{< copyable "shell-regular" >}}
+
+        ```shell
+        curl -X POST http://127.0.0.1:2379/pd/api/v1/store/<store-id>/state?state=Up
+        ```
+
+4. 假如某个 TiKV Pod 所对应的所有 Store 都处于 `Down` 或 `Tombstone` 状态，则表明异常下线已经完成。此时，需要重建 Pod 并绑定新的 PV（PersistentVolume）进行恢复：
```
Comment (weekface) [inline dev/tidb-in-kubernetes/troubleshoot.md]: 处于 Down 状态的应该不能直接删除，跟 PD 开发确认一下？

Comment (aylei) [reply dev/tidb-in-kubernetes/troubleshoot.md]: 这里写错了，应该说检查lastmodified time最新的 Store 是否处于 Tombstone 状态

File: dev/tidb-in-kubernetes/troubleshoot.md [hunk 8]
```diff
@@ -259,3 +259,69 @@ kubectl logs -f <tidb-pod-name> -n <namespace> -c tidb
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl get -n <namespace> po -l app.kubernetes.io/component=tikv
+    ```
+
+3. 对比 Store 状态与 Pod 运行状态。正常情况下每一个正常运行的 Pod 都有一个处于 `UP` 状态的 Store，假如某个 TiKV Pod 所对应的 Store 处于 `Offline` 状态，则表明该 Pod 的 Store 正在异常下线中。此时，可以通过下面的命令取消下线进程，进行恢复：
+
+    * 打开到 PD 服务的连接：
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/troubleshoot.md]: suggest using ordered list

File: dev/tidb-in-kubernetes/troubleshoot.md [hunk 9]
```diff
@@ -259,3 +259,69 @@ kubectl logs -f <tidb-pod-name> -n <namespace> -c tidb
...
+
+        {{< copyable "shell-regular" >}}
+
+        ```shell
+        curl -X POST http://127.0.0.1:2379/pd/api/v1/store/<store-id>/state?state=Up
+        ```
+
+4. 假如某个 TiKV Pod 所对应的 `lastHeartbeatTime` 最新的 Store 处于 `Tombstone` 状态 ，则表明异常下线已经完成。此时，需要重建 Pod 并绑定新的 PV（PersistentVolume）进行恢复：
+
+    * 将该 Store 对应 PV 的 `reclaimPolicy` 调整为 `Delete`：
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/troubleshoot.md]: suggest using ordered list

File: v3.0/tidb-in-kubernetes/troubleshoot.md [hunk 10]
```diff
@@ -260,3 +260,69 @@ kubectl logs -f <tidb-pod-name> -n <namespace> -c tidb
...
+
+2. 查看 TiKV Pod 运行状态：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl get -n <namespace> po -l app.kubernetes.io/component=tikv
+    ```
+
+3. 对比 Store 状态与 Pod 运行状态。正常情况下每一个正常运行的 Pod 都有一个处于 `UP` 状态的 Store，假如某个 TiKV Pod 所对应的 Store 处于 `Offline` 状态，则表明该 Pod 的 Store 正在异常下线中。此时，可以通过下面的命令取消下线进程，进行恢复：
```
Comment (dcalvin) [inline v3.0/tidb-in-kubernetes/troubleshoot.md]:
```suggestion
3. 对比 Store 状态与 Pod 运行状态。假如某个 TiKV Pod 所对应的 Store 处于 `Offline` 状态，则表明该 Pod 的 Store 正在异常下线中。此时，可以通过下面的命令取消下线进程，进行恢复：
```

Comment (dcalvin) [reply v3.0/tidb-in-kubernetes/troubleshoot.md]: Already described above.

**Comment by lilin90** [comment]
@aylei Please address @dcalvin's comments. Thanks!

**Comment by aylei** [comment]
@dcalvin @lilin90 I've addressed the review comments, PTAL

**Pull request merged**